language/vm/vm-runtime/src/runtime.rs

```diff
@@ -52,147 +60,165 @@ impl<'alloc> VMRuntime<'alloc> {
         }
     }
 
-    /// Determine if a transaction is valid. Will return `None` if the transaction is accepted,
-    /// `Some(Err)` if the VM rejects it, with `Err` as an error code. We verify the following
-    /// items:
-    /// 1. The signature on the `SignedTransaction` matches the public key included in the
-    ///    transaction
-    /// 2. The script to be executed is in the whitelist.
-    /// 3. Invokes `LibraAccount.prologue`, which checks properties such as the transaction has the
-    /// right sequence number and the sender has enough balance to pay for the gas. 4.
-    /// Transaction arguments matches the main function's type signature. 5. Script and modules
-    /// in the transaction pass the bytecode static verifier.
-    ///
-    /// Note: In the future. we may defer these checks to a later pass, as all the scripts we will
-    /// execute are pre-verified scripts. And bytecode verification is expensive. Thus whether we
-    /// want to perform this check here remains unknown.
-    pub fn verify_transaction(
+    pub(crate) fn load_gas_schedule(
```

Comment:
Let's chat about the gas schedule later. It seems to me that loading the gas schedule should not rely on the code cache. Currently we are only using code cache to derive the access path. We may hard code this access path in the future so that loading the gas schedule will have nothing to do with the code cache.

Reply:
> Let's chat about the gas schedule later. It seems to me that loading the gas schedule should not rely on the code cache. Currently we are only using code cache to derive the access path. We may hard code this access path in the future so that loading the gas schedule will have nothing to do with the code cache.

I would agree with you on this. I am not sure what the best way to do this is but it bothered me too that we had to rely in a strange way to the runtime. Think about this a bit as well

Reply:
Add a TODO for this?

Reply:
I don't think we need a TODO here, this code will continue to be changes in the next days and that will drive this refactor the proper way